contracts/oracle/src/contract/feed.rs

```diff
@@ -127,20 +149,442 @@ pub fn try_query_alarms<OracleBase>(
 where
     OracleBase: Currency + DeserializeOwned,
 {
-    let prices = calc_all_prices::<OracleBase>(storage, block_time)?;
-    MarketAlarms::try_query_alarms::<OracleBase>(storage, &prices)
+    let tree = SupportedPairs::load(storage)?;
+    let prices = calc_all_prices::<OracleBase>(storage, block_time, &tree)?;
+    MarketAlarms::try_query_alarms::<OracleBase>(storage, prices)
 }
 
-fn calc_all_prices<OracleBase>(
-    storage: &dyn Storage,
+fn calc_all_prices<'a, OracleBase>(
+    storage: &'a dyn Storage,
     block_time: Timestamp,
-) -> Result<Vec<SpotPrice>, ContractError>
+    tree: &'a SupportedPairs<OracleBase>,
+) -> Result<impl Iterator<Item = BasePrice<SwapGroup, OracleBase>> + 'a, ContractError>
 where
     OracleBase: Currency + DeserializeOwned,
 {
     let total_registered = Feeders::total_registered(storage)?;
     use crate::state::config::Config as OracleConfig;
     let config = OracleConfig::load(storage)?;
     let oracle = Feeds::<OracleBase>::with(config.price_config);
-    oracle.calc_all_prices(storage, block_time, total_registered)
+    oracle.all_prices_iter(storage, tree.swap_pairs_df(), block_time, total_registered)
+}
+
+struct LegCmd<OracleBase, Querier>
```

Comment:
too much code for a module. Please extract this and the next entities in sub-modules.